libs/community/langchain_community/vectorstores/milvus.py

```diff
@@ -1081,3 +1087,40 @@ def upsert(
                 "Failed to upsert entities: %s error: %s", self.collection_name, exc
             )
             raise exc
+
+    def get_documents_by_ids(self, ids: int | str | List[int | str]) -> List[Document]:
+        # Generating filtering expr for passing to query function
+        if isinstance(ids, list):
+            expr = "pk in ["
+
+            for id_value in ids:
+                expr += f"'{id_value}',"
+
+            expr += "]"
+
+        else:
+            expr = "pk in ['{id}']".format(id=ids)
```

Comment:
**Issue**: The method of constructing the query expression string is inefficient and prone to SQL injection-like issues if not properly sanitized. <br> **Fix**: Use parameterized queries or ensure proper sanitization. Additionally, refactor the string construction to be more efficient. <br> **Code Suggestion**: 
 ```
 -        if isinstance(ids, list):
 -            expr = "pk in ["
 -            for id_value in ids:
 -                expr += f"'{id_value}',"
 -            expr += "]"
 -        else:
 -            expr = "pk in ['{id}']".format(id=ids)
 +        if isinstance(ids, list):
 +            id_list = ','.join(f"'{id_value}'" for id_value in ids)
 +            expr = f"pk in [{id_list}]"
 +        else:
 +            expr = f"pk in ['{ids}']"
 ```